### PR TITLE
Added fullpath function for resolution of '~/.mass_raw.agr' when using gfortran

### PR DESCRIPTION
**Pull request by RarrugasTetra**

### CODE REVIEW

File: main.f90 [hunk 1]
```diff
@@ -1003,3 +1005,23 @@ subroutine calcfr(pp,pair,sum4)
...
+! Note, this function replaces shell tilde ~/ with the explicit home dir string.
+function fullpath(fname)
+  implicit none
+  character fname*(*)
+  character(len=4096) :: absdir, fullpath
+  if(fname(1:1) .eq. '/') then
+    fullpath = fname
+  else if(fname(1:2) .eq. '~/') then
+    call getenv('HOME', absdir)
+    fullpath = absdir(:lnblnk(absdir)) // fname(2:lnblnk(fname))
```
Comment (awvwgk) [inline main.f90:1020]: `lnblnk` is a GNU extension, please use `len_trim` instead (see https://gcc.gnu.org/onlinedocs/gfortran/LNBLNK.html)

Comment (RarrugasTetra) [reply main.f90]: Ok, no problem, I've already forgotten Fortran...

File: main.f90 [hunk 2]
```diff
@@ -1003,3 +1005,23 @@ subroutine calcfr(pp,pair,sum4)
...
+  if(fname(1:1) .eq. '/') then
+    fullpath = fname
+  else if(fname(1:2) .eq. '~/') then
+    call getenv('HOME', absdir)
+    fullpath = absdir(:len_trim(absdir)) // fname(2:len_trim(fname))
+  else
+    call getcwd(absdir)
+    fullpath = absdir(:len_trim(absdir)) // '/' // fname(:len_trim(fname))
+  endif
+  return
```
Comment (awvwgk) [inline main.f90:1025]: No need to return at the end of procedure
```suggestion
```

File: main.f90 [hunk 3]
```diff
@@ -88,6 +88,7 @@ program plotms
   character(len=80)  :: arg(10)
   character(len=80)  :: xname
   character(len=:), allocatable  :: fname,fname1,fname2,fname3,fname4
+  character(len=4096) :: fullpath
```
Comment (awvwgk) [inline main.f90:91]: Either declare it as external, or make it an internal procedure of the program.
```suggestion
  character(len=4096), external :: fullpath
```

File: main.f90 [hunk 4]
```diff
@@ -1003,3 +1005,23 @@ subroutine calcfr(pp,pair,sum4)
   enddo
   
 end subroutine calcfr
+
+
+! Get the absolute path to file
+! Note, this function replaces shell tilde ~/ with the explicit home dir string.
+function fullpath(fname)
+  implicit none
+  character fname*(*)
```
Comment (awvwgk) [inline main.f90:1014]: This syntax is deprecated.
```suggestion
  character(len=*), intent(in) :: fname
```

**Comment by RarrugasTetra** [comment]
Thank you for your help!
Actually I am not a Fortran programmer and just wanted to add the required in my work isotopes.
It is likely that I will do this for some other elements.

**Pull request merged**